### ISSUE TITLE
Pandas UserWarning: The argument 'infer_datetime_format' is deprecated

### ISSUE THREAD

**Issue by bastien-k** [gjoseph92/stackstac#213]
https://github.com/gjoseph92/stackstac/blob/e30ecf233ee42c1c98fcfbc5178ff105ba60760e/stackstac/prepare.py#L383

With pandas 2.0.3, this line returns a UserWarning:
```md
UserWarning: The argument 'infer_datetime_format' is deprecated and will be removed in a future version. 
A strict version of it is now the default,
see https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. 
You can safely remove this argument.
```

**Comment by gjoseph92** [comment]
Thanks for pointing this out @bastien-k. Would you have any interest in submitting a fix? Seems like just removing that argument and setting the minimum pandas version to 2.0.3 would do it. (Though I'm a little hesitant to require newer pandas just for that; idk if pandas 2.0 would cause issues for any users.)

**Comment by gjoseph92** [comment]
Finally fixed in 0.5.1, now available on pypi
